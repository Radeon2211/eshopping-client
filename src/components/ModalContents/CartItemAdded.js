import React from 'react';

const CartItemAdded = () => {
  return <h2>Cart item added</h2>;
};

export default CartItemAdded;
