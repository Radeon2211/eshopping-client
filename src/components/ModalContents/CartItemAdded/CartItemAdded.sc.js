import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const ProductPreview = styled.div`
  display: flex;

  & .photo-box {
    align-items: center;
    display: flex;
    justify-content: center;
    max-height: 11rem;
    margin-right: ${({ theme }) => theme.spacings.level3};
    width: 8rem;
  }

  & .photo {
    max-height: 100%;
    max-width: 100%;
    object-fit: cover;
  }

  & .data-box {
    align-items: flex-start;
    display: flex;
    flex-direction: column;
  }

  & .name {
    font-size: ${({ theme }) => theme.fontSizes.level3};
    margin-top: ${({ theme }) => theme.spacings.level1};
  }

  & .quantity {
    font-size: ${({ theme }) => theme.fontSizes.level2};
  }

  & .price {
    font-size: 2.2rem;
  }

  & .total-in-cart {
    font-size: ${({ theme }) => theme.fontSizes.level2};
  }
`;
