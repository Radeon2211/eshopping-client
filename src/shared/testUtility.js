// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const historyPageNum = (pageNumber) => ({
  listen: jest.fn(),
  location: { search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push: jest.fn(),
});

export const propsPagination = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  maxQuantityPerPage: 2,
});
