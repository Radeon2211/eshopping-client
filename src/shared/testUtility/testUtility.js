// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';
import { v4 as uuidv4 } from 'uuid';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../../store/reducers/rootReducer';
import { productConditions } from '../constants';

export const defaultDeliveryAddress = {
  firstName: 'firstName',
  lastName: 'lastName',
  street: 'street',
  zipCode: 'zipCode',
  city: 'city',
  country: 'country',
  phone: 'phone',
};

export const defaultUserProfile = {
  ...defaultDeliveryAddress,
  _id: 'userId',
  username: 'username',
  email: 'email@domain.com',
  status: 'active',
  contacts: {
    email: true,
    phone: true,
  },
  createdAt: '2021-02-10T19:10:38.872Z',
};

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const createHistoryPageNumber = (pageNumber = 1, push = jest.fn()) => ({
  listen: jest.fn(),
  location: { pathname: '/products', search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push,
});

export const createPaginationProps = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  quantityPerPage: 2,
});

export const createProductItem = (
  productId,
  sellerUsername = 'username',
  quantity = 1,
  price = 2,
  name = 'product name',
  photo = false,
  quantitySold = 0,
  buyerQuantity = 0,
  description = '',
  condition = productConditions.NEW,
) => ({
  _id: productId || uuidv4(),
  name,
  price,
  quantity,
  quantitySold,
  buyerQuantity,
  description,
  condition,
  photo,
  seller: {
    username: sellerUsername,
  },
  __v: 0,
  createdAt: '2021-02-10T19:10:38.872Z',
  updatedAt: '2021-02-10T19:10:38.872Z',
});

export const createCartItem = (
  sellerUsername = 'username',
  quantity = 1,
  productId,
  price = 2,
  name = 'product name',
  productQuantity = 2,
  photo = false,
) => {
  const product = createProductItem(
    productId || uuidv4(),
    sellerUsername,
    productQuantity,
    price,
    name,
    photo,
  );

  return {
    _id: uuidv4(),
    quantity,
    product,
  };
};

export const createTransactionAndOrderProdItem = (
  productId,
  sellerUsername = 'username',
  quantity = 1,
  price = 2,
  name = 'product name',
  photo = false,
) => ({
  _id: productId || uuidv4(),
  name,
  photo,
  price,
  quantity,
  seller: {
    username: sellerUsername,
  },
});

export const createOrder = (
  products = [createTransactionAndOrderProdItem()],
  orderId = uuidv4(),
  sellerUsername,
  buyerUsername,
  overallPrice,
  createdAt,
  sellerEmail,
  sellerPhone,
) => {
  const seller = sellerUsername
    ? {
        username: sellerUsername,
        email: sellerEmail,
        phone: sellerPhone,
      }
    : null;

  const buyer = buyerUsername
    ? {
        username: buyerUsername,
      }
    : null;

  return {
    _id: orderId,
    seller,
    buyer,
    products,
    overallPrice,
    deliveryAddress: defaultDeliveryAddress,
    createdAt,
    updatedAt: createdAt,
    __v: 0,
  };
};

export const testStore = (initialAuth = {}, initialProduct = {}, initialUI = {}) => {
  const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
  const defaultState = createStoreWithMiddleware(rootReducer).getState();

  const finalState = {
    auth: {
      ...defaultState.auth,
      ...initialAuth,
    },
    product: {
      ...defaultState.product,
      ...initialProduct,
    },
    ui: {
      ...defaultState.ui,
      ...initialUI,
    },
  };
  const finalStore = createStoreWithMiddleware(rootReducer, finalState);

  return {
    store: finalStore,
    initialState: finalStore.getState(),
  };
};

export const checkReqMethodAndURL = (moxiosInstance, method, url, reqNumber = 0) => {
  if (reqNumber) {
    const requests = moxiosInstance.requests.__items;
    const prevRequest = requests[reqNumber - 1];
    return method === prevRequest.config.method && url === prevRequest.config.url;
  }
  const lastRequest = moxiosInstance.requests.mostRecent();
  return method === lastRequest.config.method && url === lastRequest.config.url;
};

export const createExpectedState = (
  initialState,
  expectedAuth = {},
  expectedProduct = {},
  expectedUI = {},
) => {
  const newState = {
    auth: {
      ...initialState.auth,
      ...expectedAuth,
    },
    product: {
      ...initialState.product,
      ...expectedProduct,
    },
    ui: {
      ...initialState.ui,
      ...expectedUI,
    },
  };

  return newState;
};

export const setUpStoreWithDefaultProfile = (auth = {}, product = {}, ui = {}) => {
  return testStore(
    {
      profile: defaultUserProfile,
      deliveryAddress: defaultDeliveryAddress,
      cart: [],
      ...auth,
    },
    product,
    ui,
  );
};
