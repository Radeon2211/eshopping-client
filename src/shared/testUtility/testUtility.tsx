/* eslint-disable import/no-extraneous-dependencies */
import { ReactElement } from 'react';
import { Router } from 'react-router-dom';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { MockStoreEnhanced } from 'redux-mock-store';
import { fireEvent, waitFor, render } from '@testing-library/react';
import { v4 as uuidv4 } from 'uuid';
import moxios from 'moxios';
import rootReducer from '../../store/reducers/rootReducer';
import theme from '../../styled/theme';
import {
  CartItem,
  DeliveryAddress,
  Order,
  OrderProductItem,
  Product,
  Profile,
  ProfileStatus,
} from '../types/types';
import { ProductCondition } from '../types/enums';

export const defaultDeliveryAddress: DeliveryAddress = {
  firstName: 'firstName',
  lastName: 'lastName',
  street: 'street',
  zipCode: 'zipCode',
  city: 'city',
  country: 'country',
  phone: 'phone',
};

export const defaultUserProfile: Profile = {
  ...defaultDeliveryAddress,
  _id: 'userId',
  username: 'username',
  email: 'email@domain.com',
  status: ProfileStatus.ACTIVE,
  contacts: {
    email: true,
    phone: true,
  },
  createdAt: '2021-02-10T19:10:38.872Z',
};

export const createPaginationProps = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  quantityPerPage: 2,
});

export const createProductItem = (props: Partial<Product> = {}): Product => {
  return {
    _id: uuidv4(),
    quantity: 1,
    price: 2,
    name: 'product name',
    photo: false,
    quantitySold: 0,
    buyerQuantity: 0,
    description: '',
    condition: ProductCondition.NEW,
    seller: {
      username: 'username',
    },
    createdAt: '2021-02-10T19:10:38.872Z',
    updatedAt: '2021-02-10T19:10:38.872Z',
    __v: 0,
    ...props,
  };
};

export const createCartItem = (props = {}): CartItem => {
  const finalProps = {
    id: uuidv4(),
    sellerUsername: 'username',
    quantity: 1,
    productId: uuidv4(),
    price: 2,
    name: 'product name',
    productQuantity: 2,
    photo: false,
    ...props,
  };
  const product = createProductItem({
    _id: finalProps.productId,
    quantity: finalProps.productQuantity,
    price: finalProps.price,
    name: finalProps.name,
    photo: finalProps.photo,
    seller: {
      username: finalProps.sellerUsername,
    },
  });

  return {
    _id: finalProps.id,
    quantity: finalProps.quantity,
    product,
  };
};

export const createTransactionAndOrderProdItem = (
  props: Partial<OrderProductItem> = {},
): OrderProductItem => {
  return {
    _id: uuidv4(),
    quantity: 1,
    price: 2,
    name: 'product name',
    photo: false,
    seller: {
      username: 'username',
    },
    ...props,
  };
};

export const createOrder = (props: Partial<Order> = {}): Order => {
  return {
    _id: uuidv4(),
    overallPrice: 100,
    seller: null,
    buyer: null,
    products: [createTransactionAndOrderProdItem()],
    deliveryAddress: defaultDeliveryAddress,
    createdAt: '2021-02-10T19:10:38.872Z',
    updatedAt: '2021-02-10T19:10:38.872Z',
    __v: 0,
    ...props,
  };
};

export const testStore = (initialAuth = {}, initialProduct = {}, initialUI = {}) => {
  const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
  const defaultState = createStoreWithMiddleware(rootReducer).getState();

  const finalState = {
    auth: {
      ...defaultState.auth,
      ...initialAuth,
    },
    product: {
      ...(defaultState.product as object),
      ...initialProduct,
    },
    ui: {
      ...(defaultState.ui as object),
      ...initialUI,
    },
  };
  // @ts-expect-error TBF after redux upgrade
  const finalStore = createStoreWithMiddleware(rootReducer, finalState);

  return {
    store: finalStore,
    initialState: finalStore.getState(),
  };
};

type MoxiosRequestsItems = { config: { method: string; url: string } }[];
type MockedMoxiosInstance = {
  requests: {
    __items?: MoxiosRequestsItems;
    mostRecent?: () => { config: { method: string; url: string } };
  };
};
type DefaultMoxiosInstance = Omit<typeof moxios, 'requests'> & {
  requests: typeof moxios.requests & {
    __items?: MoxiosRequestsItems;
  };
};
export const checkReqMethodAndURL = (
  moxiosInstance: MockedMoxiosInstance | DefaultMoxiosInstance,
  method: string,
  url: string,
  reqNumber = 0,
) => {
  if (reqNumber) {
    const requests = moxiosInstance.requests.__items;
    const prevRequest = requests?.[reqNumber - 1];
    return prevRequest && method === prevRequest.config.method && url === prevRequest.config.url;
  }
  if (!moxiosInstance.requests.mostRecent) return false;
  const lastRequest = moxiosInstance.requests.mostRecent();
  return method === lastRequest.config.method && url === lastRequest.config.url;
};

export const createExpectedState = (
  initialState: {
    auth?: object;
    product?: object;
    ui?: object;
  },
  expectedAuth = {},
  expectedProduct = {},
  expectedUI = {},
) => {
  const newState = {
    auth: {
      ...initialState.auth,
      ...expectedAuth,
    },
    product: {
      ...initialState.product,
      ...expectedProduct,
    },
    ui: {
      ...initialState.ui,
      ...expectedUI,
    },
  };

  return newState;
};

export const setUpStoreWithDefaultProfile = (auth = {}, product = {}, ui = {}) => {
  return testStore(
    {
      profile: defaultUserProfile,
      deliveryAddress: defaultDeliveryAddress,
      cart: [],
      ...auth,
    },
    product,
    ui,
  );
};

export const clickAtSubmitButton = async (container: Element) => {
  await waitFor(() => {
    const submitBtn = container.querySelector('button[type="submit"]');
    if (submitBtn) {
      fireEvent.click(submitBtn);
    }
  });
};

export const testRouterPushCall = (
  pushFn: ReturnType<typeof jest.fn>,
  callIdx: number,
  expectedPathname: string,
  expectedSearch = '',
) => {
  expect(pushFn.mock.calls[callIdx][0].pathname).toEqual(expectedPathname);
  expect(pushFn.mock.calls[callIdx][0].search).toEqual(expectedSearch);
};

type RenderAppPartOptions = {
  push?: ReturnType<typeof jest.fn>;
  pathname?: string;
  search?: string;
  withoutThemeProvider?: boolean;
  withoutRouter?: boolean;
  store?: MockStoreEnhanced<object, object>;
  onlyReturnWrappedElement?: boolean;
};

export const renderAppPart = (element: ReactElement, options: RenderAppPartOptions = {}) => {
  const navigator = {
    createHref: jest.fn(),
    push: options.push || jest.fn(),
    go: jest.fn(),
    replace: jest.fn(),
  };

  const location = {
    pathname: options.pathname || '/',
    search: options.search || '',
  };

  let content = element;
  if (!options.withoutThemeProvider) {
    content = <ThemeProvider theme={theme}>{element}</ThemeProvider>;
  }
  if (options.store) {
    content = <Provider store={options.store}>{content}</Provider>;
  }
  if (!options.withoutRouter) {
    content = (
      <Router location={location} navigator={navigator}>
        {content}
      </Router>
    );
  }

  return options.onlyReturnWrappedElement ? content : render(content);
};
