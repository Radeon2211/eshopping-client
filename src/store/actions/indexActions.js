export {
  registerUser,
  loginUser,
  fetchProfile,
  logoutUser,
  fetchOtherUser,
  setOtherUser,
  updateUser,
  changeDeliveryAddress,
  deleteAccount,
  addAdmin,
  removeAdmin,
  sendAccountVerificationLink,
  resetPassword,
} from './authActions';

export {
  fetchCart,
  addCartItem,
  updateCartItem,
  clearCart,
  removeCartItem,
  setTransaction,
  goToTransaction,
  buyProducts,
} from './tradeActions';

export { setModal, deleteMessage, changeProductsPerPage } from './uiActions';

export {
  addProduct,
  editProduct,
  fetchProducts,
  fetchProductDetails,
  setProductDetails,
  deleteProduct,
} from './productActions';

export { fetchOrders, fetchOrderDetails, setOrderDetails } from './orderActions';
