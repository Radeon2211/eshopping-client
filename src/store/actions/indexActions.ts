export {
  registerUser,
  loginUser,
  fetchProfile,
  logoutUser,
  fetchOtherUser,
  setOtherUser,
  updateUser,
  changeEmail,
  changeDeliveryAddress,
  deleteAccount,
  addAdmin,
  removeAdmin,
  sendAccountVerificationLink,
  resetPassword,
} from './authActions/authActions';

export {
  fetchCart,
  addCartItem,
  updateCartItem,
  clearCart,
  removeCartItem,
  setTransaction,
  goToTransaction,
  buyProducts,
} from './tradeActions/tradeActions';

export { setMessage, setModal, changeProductsPerPage } from './uiActions/uiActions';

export {
  addProduct,
  editProduct,
  fetchProducts,
  fetchProductDetails,
  setProductDetails,
  deleteProduct,
} from './productActions/productActions';

export { fetchOrders, fetchOrderDetails, setOrderDetails } from './orderActions/orderActions';
