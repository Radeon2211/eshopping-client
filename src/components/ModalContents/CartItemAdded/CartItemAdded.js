import { useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import * as actions from '../../../store/actions/indexActions';
import Button from '../../UI/Button/Button';
import Heading from '../../UI/Heading/Heading';
import FlexWrapper from '../../UI/FlexWrapper';
import PlainText from '../../UI/PlainText';
import Loader from '../../UI/Loader/Loader';
import { formatPrice } from '../../../shared/utility/utility';
import ProductThumbnail from '../../UI/ProductThumbnail/ProductThumbnail';

export default function CartItemAdded() {
  const productDetails = useSelector((state) => state.product.productDetails);
  const cart = useSelector((state) => state.auth.cart);
  const isCartLoading = useSelector((state) => state.ui.isCartLoading);

  const dispatch = useDispatch();
  const onSetModal = useCallback(
    (modalContent) => dispatch(actions.setModal(modalContent)),
    [dispatch],
  );

  const addedProductInCart = cart.find((item) => item?.product?._id === productDetails?._id);

  let content = <Loader />;
  if (!isCartLoading && addedProductInCart) {
    const {
      quantity,
      product: { _id: productId, name, price, photo },
    } = addedProductInCart;

    const formattedPrice = formatPrice(price);

    content = (
      <>
        <Heading $variant="h3" $align="center" $mgBottom="3">
          Product added to cart
        </Heading>
        <FlexWrapper $spacing="3">
          <ProductThumbnail photo={photo} alt={name} productId={productId} width="8" height="8" />
          <FlexWrapper $direction="column" $align="start" $spacing="1">
            <PlainText $size="5" data-testid="CartItemAdded-price-and-quantity">
              {formatPrice(price * quantity)}
              <PlainText $size="2">
                &nbsp;
                {`(total in the cart ${quantity} x ${formattedPrice})`}
              </PlainText>
            </PlainText>
            <PlainText $size="3" data-testid="CartItemAdded-quantity-and-name">
              <PlainText $size="2">{`${quantity} x`}</PlainText>
              &nbsp;
              {name}
            </PlainText>
          </FlexWrapper>
        </FlexWrapper>
        <FlexWrapper $mgTop="3" $justify="center" $spacing="3">
          <Button $color="blue" clicked={() => onSetModal('')}>
            Continue shopping
          </Button>
          <Link to="/cart">
            <Button $color="blue" $filled clicked={() => onSetModal('')}>
              Go to cart
            </Button>
          </Link>
        </FlexWrapper>
      </>
    );
  }

  return content;
}
