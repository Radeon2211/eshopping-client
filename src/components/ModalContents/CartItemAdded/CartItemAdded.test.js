import React from 'react';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import CartItemAdded from './CartItemAdded';
import theme from '../../../styled/theme';
import { createProductItem } from '../../../shared/testUtility/testUtility';
import * as actions from '../../../store/actions/indexActions';

const mockStore = configureMockStore([thunk]);

const createStore = (cart, productDetails, isCartLoading) => {
  const store = mockStore({
    auth: { cart },
    product: { productDetails },
    ui: { isCartLoading },
  });
  store.dispatch = jest.fn();
  return store;
};

const defaultProduct = createProductItem({
  id: 'p1',
  sellerUsername: 'user1',
  price: 500.6,
  quantity: 4,
  name: 'product1',
  photo: true,
});

const defaultCart = [{ _id: 'item1', quantity: 3, product: defaultProduct }];

const setUp = (store, pushFn = jest.fn()) => {
  const history = {
    listen: jest.fn(),
    createHref: jest.fn(),
    location: { pathname: '/product/p1' },
    push: pushFn,
  };

  return render(
    <Provider store={store}>
      <Router history={history}>
        <ThemeProvider theme={theme}>
          <CartItemAdded />
        </ThemeProvider>
      </Router>
    </Provider>,
  );
};

afterEach(cleanup);

describe('<CartItemAdded />', () => {
  describe('Check how renders', () => {
    it('should render everything correctly if isCartLoading is false', () => {
      const store = createStore(defaultCart, defaultProduct);
      const { asFragment } = setUp(store);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render only <Loader /> if isCartLoading is true', () => {
      const store = createStore(defaultCart, defaultProduct, true);
      const { asFragment } = setUp(store);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render only <Loader /> if given product is not in cart', () => {
      const store = createStore([], defaultProduct, true);
      const { asFragment } = setUp(store);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('Check reactions to buttons clicks', () => {
    it('should call push with /cart after go to cart button click', () => {
      const store = createStore(defaultCart, defaultProduct);
      const pushFn = jest.fn();
      setUp(store, pushFn);

      fireEvent.click(screen.getByTestId('CartItemAdded-cart-link'));
      expect(pushFn).toHaveBeenCalledWith('/cart');
      expect(pushFn).toHaveBeenCalledTimes(1);
    });

    it('should call setModal() after buttons clicks', () => {
      const store = createStore(defaultCart, defaultProduct);
      const pushFn = jest.fn();
      setUp(store, pushFn);

      expect(store.dispatch).not.toHaveBeenCalled();
      fireEvent.click(screen.getByText('Continue shopping'));
      expect(store.dispatch).toHaveBeenNthCalledWith(1, actions.setModal(false));
      fireEvent.click(screen.getByText('Go to cart'));
      expect(store.dispatch).toHaveBeenNthCalledWith(2, actions.setModal(false));
      expect(store.dispatch).toHaveBeenCalledTimes(2);
    });
  });
});
