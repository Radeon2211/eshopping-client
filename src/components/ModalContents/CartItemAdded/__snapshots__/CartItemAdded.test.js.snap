// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItemAdded /> check how renders should render everything correctly if product is in cart and isCartLoading is false 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 2.4rem;
  font-family: Noto Sans,sans-serif;
  line-height: 1;
  margin-bottom: 2.4rem;
  margin-top: 0;
  text-align: center;
  margin-bottom: 2.4rem;
}

<h3
    class="c0"
  >
    Product added to cart
  </h3>
  .c4 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  font-size: 2.2rem;
}

.c5 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  font-size: 1.4rem;
}

.c6 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  font-size: 1.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
}

.c0 > *:not(:last-child) {
  margin-right: 2.4rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 8;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > *:not(:last-child) {
  margin-right: 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c3 > *:not(:last-child) {
  margin-bottom: 0.8rem;
}

.c2 {
  max-height: 8rem;
  min-width: 8rem;
  width: 8rem;
}

.c2 img {
  max-height: 8rem;
  max-width: 8rem;
  object-fit: cover;
}

<div
    class="c0"
    direction="row"
    spacing="3"
    width="100%"
    wrap="nowrap"
  >
    <div
      class="c1 c2"
      data-testid="ProductThumbnail"
      direction="row"
      height="8"
      spacing=""
      width="8"
      wrap="nowrap"
    >
      <img
        alt="product1"
        data-testid="ProductThumbnail-img"
        src="http://192.168.1.109:4000/products/p1/photo"
      />
    </div>
    <div
      class="c3"
      direction="column"
      spacing="1"
      width="100%"
      wrap="nowrap"
    >
      <span
        class="c4"
        color=""
        display="inline-block"
        overflow="visible"
        size="5"
        spacing="0px"
        transform=""
      >
        $1,501.80
        <span
          class="c5"
          color=""
          display="inline-block"
          overflow="visible"
          size="2"
          spacing="0px"
          transform=""
        >
           (total in the cart 3 x $500.60)
        </span>
      </span>
      <span
        class="c6"
        color=""
        display="inline-block"
        overflow="visible"
        size="3"
        spacing="0px"
        transform=""
      >
        <span
          class="c5"
          color=""
          display="inline-block"
          overflow="visible"
          size="2"
          spacing="0px"
          transform=""
        >
          3 x
        </span>
         product1
      </span>
    </div>
  </div>
  .c1 {
  box-shadow: 0 0 3px rgba(0,0,0,0.03);
  border-radius: 1px;
  cursor: pointer;
  font-size: 1.4rem;
  font-weight: 700;
  line-height: 1.2;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  outline: none;
  padding: 0.8rem 1.2rem;
  position: relative;
  text-align: center;
  text-transform: uppercase;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
  background-color: transparent;
  border: 2px solid #1E8BC3;
  color: #1E8BC3;
}

.c1:hover {
  border-color: #22A7F0;
  color: #22A7F0;
}

.c2 {
  box-shadow: 0 0 3px rgba(0,0,0,0.03);
  border-radius: 1px;
  cursor: pointer;
  font-size: 1.4rem;
  font-weight: 700;
  line-height: 1.2;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  outline: none;
  padding: 0.8rem 1.2rem;
  position: relative;
  text-align: center;
  text-transform: uppercase;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
  background-color: #1E8BC3;
  border: 2px solid #1E8BC3;
  color: #fff;
}

.c2:hover {
  background-color: #22A7F0;
  border-color: #22A7F0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 2.4rem;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > *:not(:last-child) {
  margin-right: 2.4rem;
}

<div
    class="c0"
    direction="row"
    spacing="3"
    width="100%"
    wrap="nowrap"
  >
    <button
      class="c1"
      color="blue"
      type="button"
    >
      Continue shopping
    </button>
    <a>
      <button
        class="c2"
        color="blue"
        type="button"
      >
        Go to cart
      </button>
    </a>
  </div>
</DocumentFragment>
`;

exports[`<CartItemAdded /> check how renders should render only <Loader /> if given product is not in cart 1`] = `
<DocumentFragment>
  .c0 {
  display: inline-block;
  position: relative;
  width: 4rem;
  height: 4rem;
}

.c0 > div {
  box-sizing: border-box;
  display: block;
  position: absolute;
  border-radius: 50%;
  -webkit-animation: rotate 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  animation: rotate 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  width: 4rem;
  height: 4rem;
  border: 0.5rem solid #1E8BC3;
  border-color: #1E8BC3 transparent transparent transparent;
}

.c0 > div:nth-child(1) {
  -webkit-animation-delay: -0.45s;
  animation-delay: -0.45s;
}

.c0 > div:nth-child(2) {
  -webkit-animation-delay: -0.3s;
  animation-delay: -0.3s;
}

.c0 > div:nth-child(3) {
  -webkit-animation-delay: -0.15s;
  animation-delay: -0.15s;
}

<div
    class="c0"
    data-testid="Loader"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</DocumentFragment>
`;

exports[`<CartItemAdded /> check how renders should render only <Loader /> if isCartLoading is true 1`] = `
<DocumentFragment>
  .c0 {
  display: inline-block;
  position: relative;
  width: 4rem;
  height: 4rem;
}

.c0 > div {
  box-sizing: border-box;
  display: block;
  position: absolute;
  border-radius: 50%;
  -webkit-animation: rotate 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  animation: rotate 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  width: 4rem;
  height: 4rem;
  border: 0.5rem solid #1E8BC3;
  border-color: #1E8BC3 transparent transparent transparent;
}

.c0 > div:nth-child(1) {
  -webkit-animation-delay: -0.45s;
  animation-delay: -0.45s;
}

.c0 > div:nth-child(2) {
  -webkit-animation-delay: -0.3s;
  animation-delay: -0.3s;
}

.c0 > div:nth-child(3) {
  -webkit-animation-delay: -0.15s;
  animation-delay: -0.15s;
}

<div
    class="c0"
    data-testid="Loader"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</DocumentFragment>
`;
